### PR TITLE
Cache static endpoints

### PR DESCRIPTION
**Pull request by eatonphil**

### CODE REVIEW

File: src/api/cache.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  }
+}
+
+export function set(pluralName, resources) {
+  const now = new Date();
+  setStorage(key(pluralName), {
+    ...resources,
+    __cacheUntil: new Date(now.getTime() + STATIC_ENDPOINT_CACHE_MINUTES * 60 * 1000),
+  });
+}
```
Comment (mparke) [inline src/api/cache.js]: 😍

File: src/api/cache.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+export function get(pluralName) {
+  const resource = getStorage(key(pluralName));
+  if (!resource) {
+    return;
+  }
+
+  const cacheValid = new Date(resource.__cacheUntil) > new Date();
+  if (cacheValid) {
+    delete resource.__cacheUntil;
```
Comment (mparke) [inline src/api/cache.js]: this invalidates if it's currently valid?

should this be inverted?

Comment (eatonphil) [inline src/api/cache.js]: No this doesn't invalidate anything. It just removes a key on the way out that we entered on the way in.

Comment (mparke) [inline src/api/cache.js]: the way it reads, if valid, delete the key that makes it valid and return.

so, right now if I called this twice in a row what happens? does it behave like:
```
// assuming some initial set
cache.get('key') // returns resource
cache.get('key') // returns undefined
```

Comment (eatonphil) [inline src/api/cache.js]: It doesn't delete anything from localstorage, so no that would not happen.

Comment (mparke) [inline src/api/cache.js]: I see now, thanks

**Comment by mparke** [comment]
Should we add a mechanism to force clearing of the cache, should we need to deploy updates that are coupled to db changes? 

30 minutes is pretty short, but you could still create a break for the user without clearing, in which case even 1 minute of broken is too long

perhaps this could be done by always forcing clear of major semver change.

thoughts?

**Comment by mparke** [comment]
@eatonphil want to mark DO NOT MERGE for now?

**Pull request closed without merging**